shell/components/Questions/Reference.vue

```diff
@@ -90,15 +117,17 @@ export default {
     class="row"
   >
     <div class="col span-6">
-      <LabeledSelect
-        :mode="mode"
-        :options="options"
+      <ResourceLabeledSelect
+        :resourceType="typeName"
+        :inStore="inStore"
         :disabled="$fetchState.pending || disabled"
         :label="displayLabel"
         :placeholder="question.description"
         :required="question.required"
         :value="value"
         :tooltip="displayTooltip"
+        :paginated-resource-settings="paginateResourceSetting"
+        :all-resources-settings="allResourceSettings"
```

Comment:
By any chance do we want to keep the interface consistent with PaginatedResourceTable? i.e. `api-filter` and `local-filter`.

Reply:
when i stumbled over this again i had the same thought.

i'll update cover
- prop names
- make a common type to fetch / filter the resources

Reply:
went full circle. 

the prop names here better describe what the control is doing (showing all in a dropdown or showing pages in a dropdown), the function calls are also different in nature.

did tidy it up though and move the types into their own file, added jsdocs, etc https://github.com/rancher/dashboard/pull/13025/commits/c19aabe62ce1d8a1eba6621a4bbbadfda38d539d